typing/typedecl.ml

```diff
@@ -671,7 +990,9 @@ let check_well_founded env loc path to_check ty =
 
 let check_well_founded_manifest env loc path decl =
   if decl.type_manifest = None then () else
-  let args = List.map (fun _ -> Ctype.newvar()) decl.type_params in
+  let args =
+    List.map (fun _ -> Ctype.newvar Type_layout.any) decl.type_params
```

Comment:
I've left a comment saying they shouldn't matter for `check_well_founded`, which I think is true.  In general putting the correct layouts here isn't completely trivial for two reasons: `decl.type_params` can be arbitrarily complex `type_exprs` because of constraints (I think?) and anyway this is called before we've finalized the layouts of everything.  The latter problem is unique to `type_decl`, but the former problem happens all over the place.